### PR TITLE
Add swipable and favorite feature to event

### PR DESCRIPTION
**Pull request by hugolan**

### CODE REVIEW

File: app/src/main/java/com/github/onedirection/navigation/fragment/home/DisplayEvent.java [hunk 1]
```diff
@@ -123,6 +134,27 @@ public void buttonStartDeleteEvent(View view){
               }));
     }
 
+    /** Called when the user taps the star button */
+    @SuppressLint("UseCompatLoadingForDrawables")
```
Comment (TheImpressiveDonut) [inline app/src/main/java/com/github/onedirection/navigation/fragment/home/DisplayEvent.java:138]: NOOO ! I think you use something like ContextCompat.getRessources(context, id) or something like that.

**Review by TheImpressiveDonut — CHANGES_REQUESTED** [review]
Looks good !

File: app/src/main/java/com/github/onedirection/navigation/fragment/home/DisplayEvent.java [hunk 2]
```diff
@@ -89,13 +91,22 @@ protected void onCreate(Bundle savedInstanceState) {
         startTime.setText(event.getStartTime().format(formatter));
         TextView endTime = this.findViewById(R.id.eventEndTimeDisplay);
         endTime.setText(event.getEndTime().format(formatter));
+
+        if(!HomeFragment.homeFragment.favorites.containsKey(event.getId())){
+            HomeFragment.homeFragment.favorites.put(event.getId(),false);
+        }
+        if(HomeFragment.homeFragment.favorites.get(event.getId())){
+            ImageButton btn = (ImageButton)findViewById(R.id.favorite_button);
+            btn.setImageDrawable(getResources().getDrawable(android.R.drawable.btn_star_big_on));
```
[the remark this answers is not in the document]
Comment (hugolan) [reply app/src/main/java/com/github/onedirection/navigation/fragment/home/DisplayEvent.java:100]: done

File: app/src/main/java/com/github/onedirection/navigation/fragment/home/DisplayEvent.java (hunk 1 above)
Comment (hugolan) [reply app/src/main/java/com/github/onedirection/navigation/fragment/home/DisplayEvent.java:138]: did fixes

**Review by TheImpressiveDonut — APPROVED** [review]

**Review by Flechman — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/github/onedirection/database/queries/EventQueries.java [hunk 3]
```diff
@@ -282,8 +282,9 @@ public EventQueries(Database db) {
                                 return CompletableFuture.completedFuture(e.getId()); //The event has been removed when changing the recurrence end time, so it is not modified in the database
                             }
                         }
-                        return e.equals(event) ? CompletableFuture.completedFuture(e.getId())
-                                : db.store(new Event(e.getId(), event.getName(), event.getLocationName(), event.getCoordinates(), event.getStartTime(), event.getEndTime(), e.getRecurrence()));
+                        Event new_event = new Event(e.getId(), event.getName(), event.getLocationName(), event.getCoordinates(), event.getStartTime(), event.getEndTime(), e.getRecurrence());
+                        return e.equals(new_event) ? CompletableFuture.completedFuture(new_event.getId())
+                                : db.store(new_event);
```
[the remark this answers is not in the document]
Comment (hugolan) [reply app/src/main/java/com/github/onedirection/database/queries/EventQueries.java:287]: this is for future is we want want to add the favorites to the db we should do it around here

[the remark this answers is not in the document]
Comment (Flechman) [reply app/src/main/java/com/github/onedirection/database/queries/EventQueries.java:287]: You could restore the changes for the moment

**Review by Flechman — APPROVED** [review]
Changes requested should be taken into account either now or in your next pr

**Pull request merged**